website/views.py

```diff
@@ -228,122 +280,56 @@ def _build_guid_url(base, suffix=None):
-        referent = guid_object.referent
-        if referent is None:
-            logger.error('Referent of GUID {0} not found'.format(guid))
-            raise HTTPError(http_status.HTTP_404_NOT_FOUND)
-        if not referent.deep_url:
-            raise HTTPError(http_status.HTTP_404_NOT_FOUND)
-
-        # Handle file `/download` shortcut with supported types.
-        if suffix and suffix.rstrip('/').lower() == 'download':
-            file_referent = None
-            if isinstance(referent, Preprint) and referent.primary_file:
-                file_referent = referent.primary_file
-            elif isinstance(referent, BaseFileNode) and referent.is_file:
-                file_referent = referent
-
-            if file_referent:
-                if isinstance(file_referent.target, Preprint) and not file_referent.target.is_published:
-                    # TODO: Ideally, permissions wouldn't be checked here.
-                    # This is necessary to prevent a logical inconsistency with
-                    # the routing scheme - if a preprint is not published, only
-                    # admins and moderators should be able to know it exists.
-                    auth = Auth.from_kwargs(request.args.to_dict(), {})
-                    # Check if user isn't a nonetype or that the user has admin/moderator/superuser permissions
-                    if auth.user is None or not (auth.user.has_perm('view_submissions', file_referent.target.provider) or
-                            file_referent.target.has_permission(auth.user, permissions.ADMIN)):
-                        raise HTTPError(http_status.HTTP_404_NOT_FOUND)
-
-                # Extend `request.args` adding `action=download`.
-                request.args = request.args.copy()
-                request.args.update({'action': 'download'})
-                # Do not include the `download` suffix in the url rebuild.
-                url = _build_guid_url(unquote(file_referent.deep_url))
-                return proxy_url(url)
-
-        # Handle Ember Applications
-        if isinstance(referent, Preprint):
-            if referent.provider.domain_redirect_enabled:
-                # This route should always be intercepted by nginx for the branded domain,
-                # w/ the exception of `<guid>/download` handled above.
-                return redirect(referent.absolute_url, http_status.HTTP_301_MOVED_PERMANENTLY)
-
-            if PROXY_EMBER_APPS:
-                resp = requests.get(EXTERNAL_EMBER_APPS['preprints']['server'], stream=True, timeout=EXTERNAL_EMBER_SERVER_TIMEOUT)
-                return Response(stream_with_context(resp.iter_content()), resp.status_code)
-
-            return send_from_directory(preprints_dir, 'index.html')
-
-        # Handle DraftNodes - these should never be accessed directly
-        if isinstance(referent, DraftNode):
-            raise HTTPError(http_status.HTTP_404_NOT_FOUND)
+        resource = Guid.objects.get(_id=guid.lower()).referent
```

Comment:
Aside: I was going to say "There exists one Guid with uppercase characters. There used to be many more, but this one had a collision with its lowercase counterpart," but it looks like that object has since been deleted by its creator. Carry on.